core/cap-0070.md

```diff
@@ -0,0 +1,141 @@
+Protocol version: TBD
+```
+
+## Simple Summary
+
+This CAP introduces ledger configuration settings allowing the Stellar network to dynamically adjust ledger close times, nomination timeouts,
+and ballot timeouts to facilitate controlled, incremental improvements to block time performance.
+
+## Motivation
+
+Currently, Stellar's ledger close time and SCP timeout settings are hardcoded. This restricts the network's ability to gradually test and implement
+shorter block times, limiting improvements in performance and responsiveness. Additionally, as other network parameters such as ledger limits and transaction
+throughput evolve, SCP nomination and ballot phase timings may need to be adjusted to maintain efficiency and resilience.
+
+This CAP addresses these issues by making these parameters configurable, enabling small, incremental adjustments and performance tuning.
+
+## Goals Alignment
+
+The proposal aligns with Stellar’s goals of scalability, resilience, and performance by enabling iterative improvements in ledger close time and consensus efficiency.
+
+## Abstract
+
+We define new network configuration settings for:
+
+- Target ledger close time (milliseconds)
+- Initial nomination timeout (milliseconds)
+- Initial ballot timeout (milliseconds)
+- Per-round nomination timeout increment (milliseconds)
+- Per-round ballot timeout increment (milliseconds)
+
+These settings are introduced as ledger config parameters, allowing the Stellar network to adjust consensus timings incrementally.
+
+## Specification
+
+### XDR Changes
+
+```diff
+--- a/Stellar-contract-config-setting.x
++++ b/Stellar-contract-config-setting.x
+@@ -302,6 +302,14 @@ struct EvictionIterator {
+     uint64 bucketFileOffset;
+ };
+ 
++struct ConfigSettingSCPTiming {
++    uint32 ledgerTargetCloseTimeMilliseconds;
++    uint32 nominationTimeoutInitialMilliseconds;
++    uint32 nominationTimeoutIncrementMilliseconds;
++    uint32 ballotTimeoutInitialMilliseconds;
++    uint32 ballotTimeoutIncrementMilliseconds;
++};
++
+ // limits the ContractCostParams size to 20kB
+ const CONTRACT_COST_COUNT_LIMIT = 1024;
+ 
+@@ -325,7 +333,8 @@ enum ConfigSettingID
+     CONFIG_SETTING_LIVE_SOROBAN_STATE_SIZE_WINDOW = 12,
+     CONFIG_SETTING_EVICTION_ITERATOR = 13,
+     CONFIG_SETTING_CONTRACT_PARALLEL_COMPUTE_V0 = 14,
+-    CONFIG_SETTING_CONTRACT_LEDGER_COST_EXT_V0 = 15
++    CONFIG_SETTING_CONTRACT_LEDGER_COST_EXT_V0 = 15,
++    CONFIG_SETTING_SCP_TIMING = 16
+ };
+ 
+ union ConfigSettingEntry switch (ConfigSettingID configSettingID)
+@@ -362,5 +371,7 @@ case CONFIG_SETTING_CONTRACT_PARALLEL_COMPUTE_V0:
+     ConfigSettingContractParallelComputeV0 contractParallelCompute;
+ case CONFIG_SETTING_CONTRACT_LEDGER_COST_EXT_V0:
+     ConfigSettingContractLedgerCostExtV0 contractLedgerCostExt;
++case CONFIG_SETTING_SCP_TIMING:
++    ConfigSettingSCPTiming contractSCPTiming;
+ };
+```
+
+## Semantics
+
+- **ledgerTargetCloseTimeMilliseconds**
```

Comment:
I think this CAP should also define the min/max valid ranges for these.

Reply:
I can put in like rough min/max based on my intuition, but actual min/maxes will require some significant experimentation.